gobblin-core/src/main/java/gobblin/publisher/BaseDataPublisher.java

```diff
@@ -77,11 +85,14 @@ public void initialize() throws IOException {
           ConfigurationKeys.LOCAL_FS_URI));
       this.fss.add(FileSystem.get(uri, conf));
     }
+    this.parallelRunnerThreads =
```

Comment:
Seems most code in `intialize` can be done in the constructor.  I suggest moving all of them to the constructor.